controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -445,6 +440,12 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
 		return ctrl.Result{}, errors.Wrap(err, "failed to update CoreDNS deployment")
 	}
 
+	// Reconcile certificate expiry for machines that don't have the expiry annotation on KubeadmConfig yet.
+	// Note: This should be at the end of the reconcile so it doesn't block remediation if some machines are unhealthy. Ref:https://github.com/kubernetes-sigs/cluster-api/issues/8691
```

Comment:
I'm ok in moving this check here, but remediation was already handled before L374 (where this check was previously located. Probably, it was simply not triggered by MHC.

With this change, we are allowing rollout to happen if a KCP machine is not reporting the certificate expiry, which is good, but unrelated to remediation.

Reply:
Maybe we can add another point to say we're moving it at the end of the reconcile so it doesn't block anything else.

Reply:
Thanks!